packages/botonic-nlp/src/dataset/input-parser.ts

```diff
@@ -0,0 +1,88 @@
+import { lstatSync, readdirSync, readFileSync } from 'fs'
+import { load as loadYaml } from 'js-yaml'
+import { extname, join } from 'path'
+
+import { unique } from '../utils/array-utils'
+import { AugmenterMap, DataAugmenter } from './data-augmenter'
+import { DefinedEntity, EntitiesParser } from './entities-parser'
+
+type InputData = {
+  class?: string
+  entities?: string[]
+  'data-augmentation'?: AugmenterMap
+  samples: string[]
+}
+
+export type Sample = { text: string; class: string; entities: DefinedEntity[] }
+
+type ParsedData = { classes: string[]; entities: string[]; samples: Sample[] }
+
+export class InputParser {
+  public readonly ALLOWED_EXTENSIONS = ['.yaml', '.yml']
+
+  constructor(private path: string) {}
+
+  parse(): ParsedData {
+    const files = this.getInputFiles()
+
+    let classes: string[] = []
+    let entities: string[] = []
+    let samples: Sample[] = []
+
+    files.forEach(filePath => {
+      const parsedData = this.parseInputFile(filePath)
+      classes = classes.concat(parsedData.classes)
+      entities = entities.concat(parsedData.entities)
+      samples = samples.concat(parsedData.samples)
+    })
+
+    classes = unique(classes)
+    entities = unique(entities)
+    samples = unique(samples)
+
+    return { classes, entities, samples }
+  }
+
+  private getInputFiles(): string[] {
+    const stat = lstatSync(this.path)
+
+    if (!stat.isDirectory()) {
+      throw new Error(`path "${this.path}" must be a directory.`)
+    }
+
+    return readdirSync(this.path)
+      .filter(fileName => this.ALLOWED_EXTENSIONS.includes(extname(fileName)))
+      .map(fileName => join(this.path, fileName))
+  }
+
+  private parseInputFile(path: string): ParsedData {
+    const inputData: InputData = loadYaml(readFileSync(path))
+    const className: string = inputData.class ?? ''
```

Comment:
this class is doing 2 things (not SRP):
* load files
* parse the files.
This will make :
* testing the parsing difficult
* the 3rd party developers won't be able to configure/extend the parsing algorithms
* the 3rd party developers won't be able to read the data from somewhere else
* you can't mock the data parsing when testing the main class

Maybe we should have a main class DataParser which loads gets the InputData and calls InputParser.parseData¡, which contains all the code of this function except the first line
 the constructor of DataParser :
```
constructor(private readonly path: string, readonly parser = new DataParser()) {
}
```